### DISCUSSION TITLE
Implementing an arrayContains feature in serverside query

### DISCUSSION THREAD

**Discussion by iamshimer** [serverpod/serverpod#3098] [Q&A] [answered]
I have a requirement to fetch a list of ClassRecords from class_records table which has a field called enrolledUsers. this  enrolledUsers will have a list of students. so i want to get the ClassRecords which has a specific id in a student id field in enrolledUsers. let's say, stuId is 5, so i want to get ClassRecords which contains id == 5 in a student in the enrolledUsers. something similar to arrayContains in firebase
![Screenshot 2024-09-01 at 11 16 15](https://github.com/user-attachments/assets/a306f80b-012d-4539-ad31-dff38492a884)

**Comment by SandPod** [comment, marked as answer]
Hey @iamshimer!

If you define a relation towards the users from the `ClassRecord` this query should already be possible to make, you can read more about relations here:
- Definining a one-to-many relation: https://docs.serverpod.dev/concepts/database/relations/one-to-many
- Query and filter one-to-many relation: https://docs.serverpod.dev/concepts/database/filter#any

Here is an example of such a query: https://github.com/serverpod/serverpod/blob/fbcdedbbda3356f0c926fbb6bb1a1e4c3954421b/tests/serverpod_test_server/test_integration/database_operations/entity_relations/one_to_many/find/where_any_test.dart#L34-L38


If you are not utilizing relations, in practice a List with a Model in it is defined as a ColumnSerializable (stored as a json blob) and the issue for extending the query interface of ColumnSerializable is defined here (add your additional requirements to this issue if it is related): https://github.com/serverpod/serverpod/issues/1591


Is your case covered either by relations or the issue referenced above? If not, please specify how it differentiates.
